src/account.rs

```diff
@@ -0,0 +1,317 @@
+    /// };
+    /// let result: Command = command_payload.into();
+    /// ```
+    impl std::convert::From<&AppendRole> for Command {
+        fn from(command_payload: &AppendRole) -> Self {
+            Command {
+                version: 1,
+                command_type: 4,
+                payload: command_payload.into(),
+            }
+        }
+    }
+
+    /// # Example
+    /// ```
+    /// # use iroha::{prelude::*, account::isi::AppendRole};
+    /// #
+    /// # let command_payload = &AppendRole {
+    /// #     account_id: Id::new("account","domain"),
+    /// #     role_name: "role".to_string(),
+    /// # };
+    /// # let result: Vec<u8> = command_payload.into();
+    /// let command_payload: AppendRole  = result.into();
+    /// ```
+    impl std::convert::From<Vec<u8>> for AppendRole {
+        fn from(command_payload: Vec<u8>) -> Self {
+            bincode::deserialize(&command_payload).expect("Failed to deserialize payload.")
+        }
+    }
+
+    #[test]
+    fn append_role_command_serialization_and_deserialization() {
+        let expected = AppendRole {
+            account_id: Id::new("account", "domain"),
+            role_name: "role".to_string(),
+        };
+        let actual: AppendRole =
+            bincode::deserialize(&bincode::serialize(&expected).unwrap()[..]).unwrap();
+        assert_eq!(expected, actual);
+    }
+
+    /// The purpose of create account command is to make entity in the system, capable of sending
+    /// transactions or queries, storing signatories, personal data and identifiers.
+    #[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+    pub struct CreateAccount {
+        pub account_id: Id,
+        pub domain_name: String,
+        pub public_key: [u8; 32],
+    }
+
+    /// # Example
+    /// ```
+    /// use iroha::{prelude::*, account::isi::CreateAccount};
+    ///
+    /// let command_payload = &CreateAccount {
+    ///     account_id: Id::new("account", "domain"),
+    ///     domain_name: "domain".to_string(),
+    ///     public_key: [63; 32],
+    /// };
+    /// let result: Vec<u8> = command_payload.into();
+    /// ```
+    impl std::convert::From<&CreateAccount> for Vec<u8> {
+        fn from(command_payload: &CreateAccount) -> Self {
+            bincode::serialize(command_payload).expect("Failed to serialize payload.")
+        }
+    }
+
+    /// # Example
+    /// ```
+    /// use iroha::{prelude::*, isi::Command, account::isi::CreateAccount};
+    ///
+    /// let command_payload = &CreateAccount {
+    ///     account_id: Id::new("account", "domain"),
+    ///     domain_name: "domain".to_string(),
+    ///     public_key: [63; 32],
+    /// };
+    /// let result: Command = command_payload.into();
+    /// ```
+    impl std::convert::From<&CreateAccount> for Command {
+        fn from(command_payload: &CreateAccount) -> Self {
+            Command {
+                version: 1,
+                command_type: 5,
```

Comment:
are the command_types really needed here?

Reply:
Yes, at least for now it is our way to define the type of ISI.

Reply:
Right now it is our way to support extension of OOB ISIs